crawl-ref/source/ability.cc

```diff
@@ -2794,6 +2800,25 @@ static spret_type _do_ability(const ability_def& abil, bool fail)
             return SPRET_ABORT;
         break;
 
+    case ABIL_ASHENZARI_CURSE:
+    {
+        fail_check();
+        bool success = false;
+        for (int i = 0; i < ENDOFPACK; ++i)
+        {
+            item_def item = you.inv[i];
+            if (item.base_type == OBJ_SCROLLS && item.sub_type == SCR_REMOVE_CURSE)
+            {
+                dec_inv_item_quantity(i, 1);
+                success = ashenzari_curse_item();
+                break;
+            }
+        }
+        if (!success)
+            mpr("You need a scroll of remove curse to do this.");
+        break;
+    }
+
```

Comment:
Should the check logic be moved elsewhere? I'm not sure where it's meant to go...